sdk/containerregistry/azure-containerregistry/azure/containerregistry/_anonymous_exchange_client.py

```diff
@@ -36,25 +36,25 @@ def __init__(self, endpoint, **kwargs):  # pylint: disable=missing-client-constr
         )
 
     def get_acr_access_token(self, challenge, **kwargs):
-        # type: (str, Dict[str, Any]) -> str
+        # type: (str, Any) -> str
         parsed_challenge = _parse_challenge(challenge)
         parsed_challenge["grant_type"] = TokenGrantType.PASSWORD
         return self.exchange_refresh_token_for_access_token(
-            None,
+            "",
             service=parsed_challenge["service"],
             scope=parsed_challenge["scope"],
             grant_type=TokenGrantType.PASSWORD,
             **kwargs
         )
 
     def exchange_refresh_token_for_access_token(
-        self, refresh_token=None, service=None, scope=None, grant_type=TokenGrantType.PASSWORD, **kwargs
+        self, refresh_token, service, scope, grant_type, **kwargs
```

Comment:
Is this a public method? Or only internal? Why were all the defaults removed?
Is there any possibility that one of these wouldn't be provided?

Reply:
- It's internal used for now. 
- Will get `Bad request` when service or scope is None.